### PR TITLE
Remove waitForMappingUpdateOnAll

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
All dynamic mapping updates are done in a dedicated atomic action and is a blocking operation, we don't need this method.

For waiting for pending tasks we have `waitNoPendingTasksOnAll()`

See also https://github.com/crate/crate/pull/14936

### CODE REVIEW

**Comment by mfussenegger** [comment]
Did you give this a couple of iterations?

The atomic update + acknowledge only ensures the other nodes got the message, but we afaik still have the Schemas/DocTableInfo cache and invalidation might lag a bit.

I'm not sure if that's actually a problem, but just to be sure we aren't making things flaky with this. Would be nice if we can remove it.

**Comment by BaurzhanSakhariev** [comment]
> Did you give this a couple of iterations?

Locally it was fine, added more iterations (will revert before merging) to some tests (but not `SqlTypeMappingTest` which is already flaky), passed on CI as well.

> The atomic update + acknowledge only ensures the other nodes got the message, but we afaik still have the Schemas/DocTableInfo cache and invalidation might lag a bit.

Thanks for pointing it out. I'm now thinking if we should invalidate cache in general whenever a node gets index update ==> whenever it acks somewhere in `IndicesClusterStateService.updateIndices`. Afaik that's an entry point for receiving an update from the master and maybe it could help with LR/`SqlTypeMappingTest` flakiness). Wdyt?

**Comment by mfussenegger** [comment]
> Thanks for pointing it out. I'm now thinking if we should invalidate cache in general whenever a node gets index update ==> whenever it acks. We can do it somewhere in IndicesClusterStateService.updateIndices. Afaik that's an entry point for receiving an update from the master and maybe it could help with logical-replication-mapping-related-tests/SqlTypeMappingTest flakiness). Wdyt?

I wouldn't spend any time on this now but instead tackle it as part of https://github.com/crate/crate/issues/11939 in a more holistic way. The Table representations we use could be more tightly integrated into the ClusterState instead of being a decoded view onto it that lags behind.

File: server/src/test/java/io/crate/integrationtests/ColumnPolicyIntegrationTest.java [hunk 1]
```diff
@@ -125,6 +126,7 @@ public void testCopyFromFileStrictTable() throws Exception {
     }
 
     @Test
+    @Repeat(iterations = 500)
```
Comment (mfussenegger) [inline server/src/test/java/io/crate/integrationtests/ColumnPolicyIntegrationTest.java:129]: Don't forget to remove it again

File: server/src/test/java/io/crate/integrationtests/PartitionedTableIntegrationTest.java [hunk 2]
```diff
@@ -1880,6 +1881,7 @@ public void testAlterNumberOfShards() throws Exception {
     }
 
     @Test
+    @Repeat(iterations = 500)
```
Comment (mfussenegger) [inline server/src/test/java/io/crate/integrationtests/PartitionedTableIntegrationTest.java:1884]: Same here

File: server/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java [hunk 3]
```diff
@@ -258,6 +260,7 @@ public void testUpdateWithArgs() throws Exception {
     }
 
     @Test
+    @Repeat(iterations = 500)
```
Comment (mfussenegger) [inline server/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java:263]: and here

**Review by mfussenegger — APPROVED** [review]

**Pull request merged**